posts/static/author/author_posts.js

```diff
@@ -33,3 +33,103 @@ function unfollow_user(id) {
         window.location.reload(true)
     });
 }
+
+function pull_github_activity(){
+    let github = document.getElementsByName("github")[0].value;
+
+    githubsplit = github.split("/");
+    githubUsername = githubsplit[githubsplit.length - 1];
+
+    fetch("https://api.github.com/users/" + githubUsername + "/events", {
+        method: 'get'
+    })
+    .then(function(response){
+        if(response.status != 304){
+            return response.json();
+        }else{
+            return Promise.reject('Github activity hasnt been modified');
+        }
+    }).then(function(data) {
+        return post_github(data);
+    }).then(function(newId) {
+        return update_last_id(newId);
+    }).then(function() {
+        window.location.reload(true);
+    });
+}
+
+function update_last_id(newId){
+    let csrf = document.getElementsByName('csrfmiddlewaretoken')[0].value;
+    var formData = new FormData();
+    formData.append("newId", newId);
+    formData.append("csrfmiddlewaretoken", csrf);
+
+    return fetch('/frontend/author/github/', {
+        method: 'post',
+        headers: {
+            "X-CSRFToken": csrf
+        },
+        body: formData
+    }).then(function(response) {
+        console.log(response);
+    });
+}
+
+function post_github(response){
+    let csrf = document.getElementsByName('csrfmiddlewaretoken')[0].value;
+    let newId = response[0]['id'];
+    let oldId = document.getElementsByName('githubLastId')[0].value;
+    
+    for (var i = 0; i < response.length; i++) {
+        if(response[i]['id'] == oldId){
+            break;
+        }
+
+        var body = create_github_post(response[i]);
+        
+        fetch('/posts/', {
+            method: 'post',
+            headers: {
+                "X-CSRFToken": csrf,
```

Comment:
Please add a todo to update to use basic auth. I have changed some things for mr worldwide and will need to update your stuff to get it working